src/drivers/intel/cavs/dmic.c

```diff
@@ -1164,8 +1164,13 @@ static int dmic_set_config(struct dai *dai, struct sof_ipc_dai_config *config)
 	/* Copy the new DMIC params to persistent.  The last request
 	 * determines the parameters.
 	 */
-	memcpy(dmic_prm[di], &config->dmic,
+	ret = memcpy_s(dmic_prm[di], sizeof(*dmic_prm[di]), &config->dmic,
 	       sizeof(struct sof_ipc_dai_dmic_params));
+	if (ret != 0) {
```

Comment:
Would it be more efficient to just `assert()` the `memcpy_s()` status instead of transporting the error code value back to the top level caller? Any issue with memcpy destination buffers seems to be _logic_error_ (=stop and fix the code) rather then _runtime_error_ (=try to handle at run-time).

Reply:
It was at first liket that - memcpy_s was in if statment, but it was not up with code standards.
request by @lgirdwood

Reply:
@dabekjakub I don't mind using assert either, these issues will always be rare and fatal.